### DISCUSSION TITLE
[Feature Request] Webhooks for reactions on pull request review comments

### DISCUSSION THREAD

**Discussion by davidkwlam** [community/community#20824] [Pull Requests]
Our GitHub app listens for webhooks on pull request review comment activity (created, updated, deleted) to avoid polling the [pull request API](https://docs.github.com/en/rest/pulls/pulls).

We'd like to receive similar webhooks for comment reactions to avoid polling the [reactions API](https://docs.github.com/en/rest/reactions). Unfortunately, it doesn't look like a [webhook](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads) exists for when someone adds a reaction.

Just off the top of my head, either one of these would work:

1. A new webhook event type for pull request review comment reactions
2. Send the existing [`pull_request_review_comment`](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request_review_comment) event when a reaction is added to or removed from a comment

Adding this would reduce the load on the API and I think many other developers would find it useful (here's a [similar thread for issue comments](https://github.com/github-community/community/discussions/7168)).

Thanks!

**Comment by elviskahoro** [comment]
Huge +1. Articulation for why this is important: People add reactions to issues as a way of expressing interest. To my knowledge there isn't a way to query the list of people who've subscribed to an issue and so instead we ask folks to add an emoji. This is better than asking them to comment because for issues with a large amount of interest, when people comment: "+1" it spams the inboxes of those currently subscribed.

**Comment by sheilalarmour** [comment]
Yes, this is correct!

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by elviskahoro** [comment]
One other aspect of this is people often upvote an issue because they're interested in it. Up voting however doesn't automatically subscribe them to the issue and so if we were to later comment and close the issue they wouldn't know it's been fixed or closed unless they explicitly subscribed in addition to upvoting.

**Comment by chngl** [comment]
+1, are there any plans for the github team to build this out? Thanks.

**Comment by waltergalvao** [comment]
+1

**Comment by whoamitty** [comment]
+1 Me too I need it badly
for manage project
and automate move issue on Done
according to reactions

**Comment by SanjaySaatyaki** [comment]
+1 I too need this since i have automatic messages generated via Bots. To know whether those are really useful or not

**Comment by TrongBYM** [comment]
+1, we need this feature as well as we want to automate surveys and polls in our GitHub pull requests!

**Comment by jwarby** [comment]
+1, we were hoping to use a count of :+1: reactions to an auto-generated PR comment to trigger another workflow but it doesn't seem to be possible :( was hoping adding/removing a reaction would trigger `issue_comment.edited`

**Comment by pape98** [comment]
+1 This is badly needed :)

**Comment by dimmyjr-nu** [comment]
+1

**Comment by suptejas** [comment]
+1 - this is a critical webhook that would be extremely useful

**Comment by tanmayshishodia** [comment]
+1 - this is a very useful feature.

**Comment by ZIJ** [comment]
+1 much needed

**Comment by pape98** [comment]
Let's just give at this point. I sadly doubt they'll ever do this 😢 😞 .

**Comment by raghavdixit99** [comment]
+1 needed

**Comment by rinormaloku** [comment]
+1 (even Slack has this)

**Comment by josem7** [comment]
+1

**Comment by elopezn20** [comment]
+1

**Comment by berrazuriz1** [comment]
+1

**Comment by v4rgas** [comment]
+1

**Comment by enzo-inc** [comment]
+1

**Comment by kellertk** [comment]
Writing +1 here because reacting to the discussion won't generate a webhook

:)

**Comment by jr-k** [comment]
+1

**Comment by nsbradford** [comment]
+1

**Comment by StefanEger** [comment]
+1

**Comment by matvejs16** [comment]
+1

**Comment by ldanilek** [comment]
+1

**Comment by zbrown001** [comment]
+1

**Comment by pTinosq** [comment]
+1

**Comment by thevolcanomanishere** [comment]
+1

**Comment by alexiscolonna** [comment]
+ 1000 that's crazy than 3 years later it still not even planned on the roadmap. 

Emojis reaction wehbooks could help githup app developers create way more interactive experience in Github.

**Comment by kyan-yang** [comment]
+1

**Comment by bracesproul** [comment]
+1

**Comment by jminuscula** [comment]
+1 We're building agentic workflows, and being able to ask for feedback easily and act on it quickly would be a huge improvement.

**Comment by BaherW** [comment]
+1

**Comment by AFMiziara** [comment]
+1

**Comment by The-MM** [comment]
+1

**Comment by shcherbachev** [comment]
+1

**Comment by tnm** [comment]
This is bonkers that this isn’t implemented.

I know exactly the stupid rails controller where this lives and how to enable this.

**Reply by pTinosq** [reply]
When the former CTO of GitHub leaves a comment like this on a feature request... damn

**Reply by elviskahoro** [reply]
@tnm ted you're probably our only hope lol I've brought this this up at like the last 3 GitHub universes

**Comment by Nzko-cyber** [comment]
Current State

Webhooks exist for comments (issue_comment, pull_request_review_comment, commit_comment, etc.), but not for reactions on those comments.

Adding/removing a reaction does not trigger any webhook.

This has been raised before in the GitHub community (including the thread you mentioned about issue comments), but so far there’s no official support.

**Comment by lannydee** [comment]
+100

**Comment by guidocelada** [comment]
+1000

**Comment by frison** [comment]
+10000

**Comment by tom-at-everyday** [comment]
+1

**Comment by asyncawaitpromise** [comment]
+1

**Comment by fnicolelli-ls** [comment]
+1

**Comment by sstraus** [comment]
+1

**Comment by Sbaia** [comment]
+1

**Comment by alliecatowo** [comment]
voting via reactions on a poll bot would be pretty dope...+1

**Comment by AtharvaSarwate** [comment]
+1

**Comment by Danielvandervelden** [comment]
+1........... how exactly hasn't this been implemented yet? 4 years guys...

**Reply by nekename** [reply]
Wow, we're here at the same time

**Reply by Danielvandervelden** [reply]
We can be sad, angry and disappointed together 🚀

**Reply by elviskahoro** [reply]
Mitchell wrote a pretty good piece on why him and Ghostty are leaving GitHub:
https://mitchellh.com/writing/ghostty-leaving-github

**Comment by ketpatil77** [comment]
This would make integrations much cleaner. Reactions are often used as lightweight signal without the notification noise of additional comments, so not emitting webhook events forces apps back into polling for something that is otherwise event-shaped. I think your second option is especially pragmatic: extend the existing pull_request_review_comment event family with reaction add/remove actions, rather than inventing a whole separate conceptual surface. That would preserve existing webhook routing patterns for clients already modeling review-comment activity.

**Comment by elviskahoro** [comment]
I think this GitHub Discussion is pretty strong evidence that GitHub's product feedback loop (at least as it relates to maintainers) is broken. I've brought this exact issue up at GitHub universe multiple times and pretty much every time someone objected and tried to come up with reasons for why it either couldn't and or hasn't been done yet. It's disappointing to say the least.

**Reply by matfax** [reply]
What's really disappointing about it is that GitHub is the inclusive hub that has made the open source community thrive. Just anecdotally, it seems to be gradually dying though. It's not just because of GitHub, and how it lacked a larger plan and perspective of how to deal with the central issues that smaller maintainers face. It's also because of AI now, and how software gradually becomes a replaceable one shot product that's just uploaded, and the user has to hop from one of those unmaintained half ready products to the next. If you look on the GitHub Actions marketplace, there's a chronic lack of maintenance and lack of trust. GitHub Actions never had a beginning and an end, as a proper product is supposed to have. It's like they threw different ideas from a brainstorming session in the same bucket and just went through with that. Certain competitors like CircleCi did a much better job with their foundational product design, but can't compete with a service that's natively integrated on GitHub, so they now show the same lack of progress. Open Source needs a place to thrive, on a platform that's maintained and that actively shapes decentralized collaboration, to prevent redundant product/library development and to guarantee maintenance even when owners don't find the interest or time. AI could be a maintenance super tool, but it won't work on a broken platform that never had such goals to begin with.

**Comment by artus9033** [comment]
Is there any chance this request will be addressed?
